build-source.sh

```diff
@@ -79,16 +79,30 @@ export GIT_COMMIT
 export GIT_BRANCH="$BRANCH_NAME"
 cd ..
 
+source_format=$(cat source/debian/source/format)
+if [[ $source_format != 3.0* ]]; then
+   echo "WARNING: Please upgrade debian source format to 3.0"
+fi
+
 source_location_file=$(sourcedebian_or_source ubports.source_location)
 if [ -n "$source_location_file" ]; then
+  echo "WARNING: ubports.source_location file is depricated, please use debian/watch file"
   while read -r SOURCE_URL && read -r SOURCE_FILENAME; do
     if [ -f "$SOURCE_FILENAME" ]; then
       rm "$SOURCE_FILENAME"
     fi
 
     wget -O "$SOURCE_FILENAME" "$SOURCE_URL"
   done <"$source_location_file"
+  is_quilt=1;
+elif [[ $source_format == "3.0 (quilt)" ]]; then
+  cd source
+  uscan --noconf --force-download --rename --download-current-version --destdir=..
```

Comment:
This probably needs extending. It may occur that a debian/copyright file has "Files-Excluded:" as on entry in the global header of the machine readable copyright file (see DEP-5 specs).

In that case, we should add the --repack option to the uscan call (and really: only if Files-Excluded is present). For most cases, we don't want to repack the orig tarball (as it breaks e.g. GPG signed tarballs).

Reply:
This could be a follow-up PR, though. I will shortly revisit the qtpim, qtfeedback and qtsystems packages in Debian and port them to uscan based Git snaptshot retrieval (currently, they have onlyjob's Git snapshot version of get-orig-source target in debian/rules). At least one of them will have a Files-Excluded section (qtsystem, iirc). You can test with that/those then.

Reply:
I have a question though: if `uscan --repack` is used, would the tarball be reproducible? I suppose that if the tarball is different, Aptly would reject the upload.

Reply:
Yeah i think we would need to do a bit more logic here if we want to support repack

Reply:
> I have a question though: if `uscan --repack` is used, would the tarball be reproducible? I suppose that if the tarball is different, Aptly would reject the upload.

Normally, when using repack, I obtain the orig tarball from the repo where I uploaded it to if I need to do follow-up uploads.

Reply:
Or I have

> > I have a question though: if `uscan --repack` is used, would the tarball be reproducible? I suppose that if the tarball is different, Aptly would reject the upload.
> 
> Normally, when using repack, I obtain the orig tarball from the repo where I uploaded it to if I need to do follow-up uploads.

Or I have the orig tarball lying around on my local filesystem, anyway...

Reply:
> Normally, when using repack, I obtain the orig tarball from the repo where I uploaded it to if I need to do follow-up uploads.

Hmm... right. And I guess in many cases we can just re-download them from our apt archive. So the priority might be:

1. Download from our apt archive (will have to account for multi-orig-tarball, and when if we had tarball in e.g. `focal_-_PR_mmm_nn` but building against `focal`).
2. Download according to `ubports.source_location`.
3. Use `uscan`.
4. Check Debian & Ubuntu archive. This step means we would prefer the same re-packed tarball with Debian/Ubuntu before re-packing it ourselves.
5. Use `uscan --repack`.

What do you think, @sunweaver and @mariogrip ?